db/users.go

```diff
@@ -83,6 +83,25 @@ func (dbc *DatabaseContext) GetPrincipal(name string, isUser bool) (info *Princi
 	return
 }
 
+func (db *DatabaseContext) DeleteRole(name string, purge bool) error {
+	authenticator := db.Authenticator()
+
+	role, err := authenticator.GetRole(name)
+	if role == nil {
+		if err == nil {
+			return base.ErrNotFound
+		}
+		return err
+	}
```

Comment:
I think this is a bit easier to reason about, as it doesn't make assumptions about role _always_ being nil when we get an error back.

```suggestion
	if err != nil {
		return err
	}

	if role == nil {
		return base.ErrNotFound
	}
```

Reply:
I did this so it matched the existing checks but happy to change it.